services/src/handlers/datasets.rs

```diff
@@ -339,7 +363,9 @@ fn suggest_main_file(upload: &Upload) -> Option<String> {
     None
 }
 
-fn auto_detect_meta_data_definition(main_file_path: &Path) -> Result<MetaDataDefinition> {
+fn auto_detect_meta_data_definition(
+    main_file_path: &Path,
+) -> Result<crate::datasets::storage::MetaDataDefinition> {
```

Comment:
shouldn't this be the MetaDataDefintion from services/api/model/servces ?

Reply:
I thought of `auto_detect_meta_data_definition` more of like an internal function which isn't directly connected to the API. Because of that I used internal types.